### ISSUE TITLE
heatmap not displaying title on load

### ISSUE THREAD

**Issue by tttp** [dc-js/dc.js#1482]
Hi,

On the heatmap, the title are empty on load: https://dc-js.github.io/dc.js/examples/heat.html

They are properly generated and displayed when you filter them, but shouldn't the title be displayed at load?

**Comment by tttp** [comment]
FYI, I "solved" the problem by removing the title and add a tooltip
```
    var tip = d3.tip()
    .attr('class', 'd3-tip wide-tip')
    .offset([-10, 0])
    .html(function (d) {
      return "<h5>"+dayFormat(d.key)+"</h5><div>"
        +d.value
        + "</div>";
    });

    graph.on('pretransition', function(chart) {
      graph.selectAll('.heat-box')
      .call(tip)
      .on('mouseover.tip', tip.show)
      .on('mouseout.tip', tip.hide);
    });

```

**Comment by gordonwoodhull** [comment]
Good catch! Thanks @tttp!

**Comment by gordonwoodhull** [comment]
Looks like this is more fallout of the port to D3v4. The code which sets the titles for both render and redraw is

https://github.com/dc-js/dc.js/blob/2fa1fca11d30a3ac170cedb9661aa7e21fc6326e/src/heatmap.js#L229-L234

On the first run, the enter selection is not automatically merged into the update selection as it used to be, so `boxes` is still empty until the merge in the following line. Moving the merge before the title updates fixes the problem.

I've added a test and a fix for 3.0.7
